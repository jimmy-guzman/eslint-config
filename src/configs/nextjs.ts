import { GLOB_NEXTJS } from "../constants";
import { nextjsRules } from "../rules/nextjs";
import { interopDefault } from "../utils/interop-default";

export default async function nextjsConfig() {
  const nextjsPlugin = await interopDefault(import("@next/eslint-plugin-next"));

  return [
    {
      files: GLOB_NEXTJS,
      name: "jimmy.codes/nextjs",
      plugins: {
        "@next/next": nextjsPlugin,
      },
      rules: await nextjsRules(),
    },
  ];
}
