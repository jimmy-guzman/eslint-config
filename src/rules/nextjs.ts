import type { Rules } from "../types";

import { interopDefault } from "../utils/interop-default";

export const nextjsRules = async () => {
  const nextjsPlugin = await interopDefault(import("@next/eslint-plugin-next"));

  return {
    ...nextjsPlugin.configs["core-web-vitals"].rules,
  } satisfies Rules;
};
