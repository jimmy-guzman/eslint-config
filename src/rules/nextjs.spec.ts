import { nextjsRules } from "./nextjs";

test("should create nextjs rules", async () => {
  await expect(nextjsRules()).resolves.toMatchSnapshot();
});
