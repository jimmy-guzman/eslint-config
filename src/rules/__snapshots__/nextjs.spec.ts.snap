// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create nextjs rules 1`] = `
{
  "@next/next/no-html-link-for-pages": "error",
  "@next/next/no-sync-scripts": "error",
}
`;
